packages/x-data-grid/src/hooks/features/columnHeaders/useGridColumnHeaders.tsx

```diff
@@ -274,6 +274,17 @@ export const useGridColumnHeaders = (props: UseGridColumnHeadersProps) => {
         computedWidth: colDef.computedWidth,
       });
 
+      const siblingWithBorderingSeparator =
+        pinnedPosition === GridPinnedColumnPosition.RIGHT
+          ? renderedColumns[i - 1]
+          : renderedColumns[i + 1];
+      const isSiblingFocused = siblingWithBorderingSeparator
+        ? columnHeaderFocus !== null &&
+          columnHeaderFocus.field === siblingWithBorderingSeparator.field
+        : false;
+      const isLastUnpinned =
+        columnIndex + 1 === columnPositions.length - 1 - (pinnedColumns.right.length - 1);
+
```

Comment:
Should the conditions above be different for LTR and RTL?

Reply:
@cherniavskii Good point, though from testing this, it looks like we've got a bigger problem to solve with pinned columns + RTL 😬 

<img width="861" alt="Screenshot 2024-09-11 at 09 31 08" src="https://github.com/user-attachments/assets/17a3f544-2832-4e9f-8482-880a246f1df4">

https://codesandbox.io/p/sandbox/rtl-pinned-columns-4k9sdh